docs/data-sources/groups.md

```diff
@@ -0,0 +1,69 @@
+---
+page_title: "infra_groups Data Source - terraform-provider-infra"
+subcategory: ""
+description: |-
+  Get a list of registered Infra groups.
+---
+
+# infra_groups
+
+Get a list of registered Infra groups.
+
+## Example Usage
+
+```terraform
+// Get all groups
+data "infra_groups" "all" {}
+
+output "my_groups" {
+  value = data.infra_groups.all.groups
+}
+
+// Get the `engineering` group
+data "infra_groups" "engineering" {
+  filter {
+    name = "Engineering"
+  }
+}
+
+// Get all groups where `admin@example.com` is a member
+data "infra_groups" "" {
+  filter {
+    user_name = "admin@example.com"
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `filter` (Block List, Max: 1) (see [below for nested schema](#nestedblock--filter))
+- `include_users` (Boolean) Include each group's members. Default is `false`.
+
+### Read-Only
+
+- `groups` (List of Object) (see [below for nested schema](#nestedatt--groups))
+- `id` (String) The ID of this resource.
+
+<a id="nestedblock--filter"></a>
+### Nested Schema for `filter`
+
+Optional:
+
+- `name` (String) Then name of the group to find.
+- `user_id` (String) The ID of the user who belongs to this group. Cannot be used with `user_name`.
+- `user_name` (String) The name of the user who belongs to this group. Cannot be used with `user_id`.
+
+
+<a id="nestedatt--groups"></a>
+### Nested Schema for `groups`
```

Comment:
This is a bug with the doc generator. It omits read-only nested field descriptions